packages/tools/fetch-tool/src/fluidFetchMessages.ts

```diff
@@ -66,6 +67,27 @@ async function* loadAllSequencedMessages(
     let requests = 0;
     let opsStorage = 0;
 
+    const teststream = deltaStorage.fetchMessages(
+        lastSeq + 1,
+        lastSeq + 2);
+
+    let statusCode;
+    let innerMostErrorCode;
+    let response;
+
+    try {
+        await teststream.read();
+    } catch (error) {
+        statusCode = error.getTelemetryProperties().statusCode;
+        innerMostErrorCode = error.getTelemetryProperties().innerMostErrorCode;
+        if (statusCode === 410 && innerMostErrorCode === "fluidDeltaDataNotAvailable") {
```

Comment:
inverting this condition would get rid of the 'else' branch.

```suggestion
        if (statusCode !== 410 || innerMostErrorCode !== "fluidDeltaDataNotAvailable") {
            throw error;
        }
        
        response = JSON.parse(error.getTelemetryProperties().response);
```